init_features.go

```diff
@@ -256,6 +259,94 @@ func (fi *featureInitializer) initFeatures(
 	return nil
 }
 
+/*  buildNamespaceMemo populates a memo to make bulk-unmarshalling and has-many unmarshalling efficient.
+ *  i.e. Don't need to do the same work for the same features class multiple times. Given:
+ *  type User struct {
+ *      Id *string
+ *      Transactions *[]Transactions `has_many:"id,user_id"`
+ *  }
+ *  type Transactions struct {
+ *      Id *string
+ *      UserId *string
+ *      Amount *float64
+ *  }
+ *  The namespace memo will be:
+ *  {
+ *      "User": {
+ *          ResolvedFieldNameToIndex: {
+ *              "id": 0,
+ *              "user.id": 0,
+ *              "transactions": 1,
+ *              "user.transactions": 1,
+ *          }
+ *      },
+ *      "Transactions": {
+ *          ResolvedFieldNameToIndex: {
+ *              "id": 0,
+ *              "transactions.id": 0,
+ *              "user_id": 1,
+ *              "transactions.user_id": 1,
+ *              "amount": 2,
+ *              "transactions.amount": 2,
+ *          }
+ *      }
+ *  }
+ */
+func (fi *featureInitializer) buildNamespaceMemo(typ reflect.Type) error {
+	if typ.Kind() == reflect.Ptr {
+		return fi.buildNamespaceMemo(typ.Elem())
+	} else if typ.Kind() == reflect.Struct && typ != reflect.TypeOf(time.Time{}) {
+		structName := typ.Name()
+		namespace := internal.ChalkpySnakeCase(structName)
+		for fieldIdx := 0; fieldIdx < typ.NumField(); fieldIdx++ {
+			fm := typ.Field(fieldIdx)
+			resolvedName, err := internal.ResolveFeatureName(fm)
+			if err != nil {
+				return errors.Wrapf(err, "error resolving feature name: %s", fm.Name)
+			}
+
+			if _, ok := fi.namespaceMemo[structName]; !ok {
+				fi.namespaceMemo[structName] = &internal.NamespaceMemoItem{}
+			}
+			nsMemo := fi.namespaceMemo[structName]
+			if nsMemo.ResolvedFieldNameToIndex == nil {
+				nsMemo.ResolvedFieldNameToIndex = map[string]int{}
+			}
+			nsMemo.ResolvedFieldNameToIndex[resolvedName] = fieldIdx
+			// Has-many features come back as a list of structs whose keys are namespaced FQNs.
+			// Here we map those keys to their respective indices in the struct, so that we
+			// don't have to do any string manipulation to deprefix the FQN when unmarshalling.
+			nsMemo.ResolvedFieldNameToIndex[namespace+"."+resolvedName] = fieldIdx
```

Comment:
Not enough context on my end here, but just want to make sure you're confident there's no possible conflict between a resolvedName `b.c` in a namespace `a` vs. a resolvedName of `a.b.c`

Reply:
thank you! good concern but resolvedName is always non-prefixed